### ISSUE TITLE
In line exemption for custom rules

### ISSUE THREAD

**Issue by nevengrgic** [looker-open-source/look-at-me-sideways#111]
I wrote the following rule to check for explores that are defined in view files, which we do not want to allow on our instance. When I try to do an exemption within one view file, the in line exemption is not recognized. I saw a similar issue with a custom rule (#50) was resolved,  but wanted to check if different custom rule configurations could still raise issues for in line exemption. Thanks!

```
#  rule: C3 {
#  description: "Explores must not exist within view files"
#  match: "$.files[*]"
#  expr_rule: 
#
#  //for files that are of type view, check that the explore key is undefined If defined, alert the developer
#  ($if
#      ($all
#          (=== ::match:$file_type "view")
#          (!== ::match:explore undefined)
#      )
#              ($concat "View file " ::match:$file_name " contains an explore. Move the explore to the appropriate model file.")
#              true
#  )
#  ;;
#  }
```

In line exemption
```
  #  LAMS
  #  rule_exemptions: {
  #   C3: "this explore is actually okay"
  #  }
```

**Comment by fabio-looker** [comment]
Hi @nevengrgic :wave: 

I vaguely recall intending to make rule ID's of the form [A-Z][0-9]* (i.e. one letter and a number) to be reserved for future LAMS use, so there may be an issue caused by that... but I will look into this and see what I find.

**Comment by nevengrgic** [comment]
Hi @fabio-looker! Thanks for the quick response. 

You are correct about the expected functionality. I also tried changing the name based on your first comment, but that didn't appear to change anything. Appreciate your help!

**Comment by fabio-looker** [comment]
Hello!

Although this was fixed in 3.0.0-beta.0 , that beta release had several issues. Many of those issues are fixed in the next beta pre-release [3.0.0-beta.1](https://www.npmjs.com/package/@looker/look-at-me-sideways/v/3.0.0-beta.1).

[v3 release notes](https://github.com/looker-open-source/look-at-me-sideways/blob/fabble/v3-extends-refinements/docs/release-notes/v3.md)

I'll keep this issue open until v3 is out of beta. Thank you for your feedback thus far!

**Comment by fabio-looker** [comment]
[v3](https://looker-open-source.github.io/look-at-me-sideways/release-notes/v3) is now generally available.

If you continue to experience any issues, please let me know. Thank you for reporting this and for your continued feedback!
